### ISSUE TITLE
Crash on iOS when GL or Metal commands run in background

### ISSUE THREAD

**Issue by collinjackson** [flutter/flutter#11639]
This issue tracks `gpus_ReturnNotPermittedKillClient` (iOS 9.x) and `MTLReleaseAssertionFailure` (on iOS 10.x). Most likely the app is being killed trying to [draw while backgrounded](https://developer.apple.com/library/content/qa/qa1766/_index.html).

Example stack trace:

```
#0    (null) in __pthread_kill ()
#1    (null) in pthread_kill ()
#2    (null) in abort ()
#3    (null) in MTLReleaseAssertionFailure ()
#4    (null) in GLDTextureRec::update(unsigned int, unsigned short*) ()
#5    (null) in GLDTextureRec::load() ()
#6    (null) in gldLoadTexture ()
#7    (null) in gliSetInteger ()
#8    (null) in gliBindViewES ()
#9    (null) in -[EAGLContext renderbufferStorage:fromDrawable:] ()
#10    (null) in 0x100520000 ()
#11    (null) in 0x100520000 ()
#12    (null) in 0x100520000 ()
#13    (null) in 0x100520000 ()
#14    (null) in __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ ()
#15    (null) in __CFRunLoopDoTimer ()
#16    (null) in __CFRunLoopDoTimers ()
#17    (null) in __CFRunLoopRun ()
#18    (null) in CFRunLoopRunSpecific ()
#19    (null) in 0x100520000 ()
#20    (null) in 0x100520000 ()
#21    (null) in _pthread_body ()
#22    (null) in _pthread_start ()
#23    (null) in thread_start ()
```

Example stack trace:

```
#0    (null) in gpus_ReturnNotPermittedKillClient ()
#1    (null) in glrKillClient ()
#2    (null) in gpusSubmitDataBuffers ()
#3    (null) in SubmitPackets(AGXContextRec*) ()
#4    (null) in 0x1004d8000 ()
#5    (null) in 0x1004d8000 ()
#6    (null) in 0x1004d8000 ()
#7    (null) in 0x1004d8000 ()
#8    (null) in __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ ()
#9    (null) in __CFRunLoopDoTimer ()
#10    (null) in __CFRunLoopDoTimers ()
#11    (null) in __CFRunLoopRun ()
#12    (null) in CFRunLoopRunSpecific ()
#13    (null) in 0x1004d8000 ()
#14    (null) in 0x1004d8000 ()
#15    (null) in _pthread_body ()
#16    (null) in _pthread_start ()
#17    (null) in thread_start ()
```

**Comment by Hixie** [comment]
We hope to have a fix for this this week.

**Comment by cbracken** [comment]
Fix has landed with flutter/flutter#11746. Please re-open if you see this again.
